testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/health/MicroProfileHealthApplicationStartupTestBase.java

```diff
@@ -0,0 +1,103 @@
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.test.integration.microprofile.health;
+
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.junit.InSequence;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.net.URL;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({MicroProfileHealthApplicationStartupSetupTask.class})
+public abstract class MicroProfileHealthApplicationStartupTestBase {
+
+    abstract void checkGlobalOutcome(ManagementClient managementClient, String operation, boolean mustBeUP, String probeName) throws IOException;
+
+    @Deployment(name = "MicroProfileHealthApplicationStartupTestBaseSetup")
+    public static Archive<?> deploySetup() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "MicroProfileHealthApplicationStartupTestBaseSetup.war")
+            .addClass(MicroProfileHealthApplicationStartupSetupTask.class);
+        return war;
+    }
+
+    @Deployment(name = "MicroProfileHealthApplicationStartupTestBase", managed = false)
+    public static Archive<?> deploy() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "MicroProfileHealthApplicationStartupTestBase.war")
+                .addClass(MyStartupProbe.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return war;
+    }
+
+    @ContainerResource
+    ManagementClient managementClient;
+
+    @ArquillianResource
+    private Deployer deployer;
+
+    @Test
+    @InSequence(1)
+    public void testApplicationStartupBeforeDeployment() throws Exception {
+        checkGlobalOutcome(managementClient, "check-started", false, null);
+
+        // deploy the archive
+        deployer.deploy("MicroProfileHealthApplicationStartupTestBase");
+    }
+
+    @Test
+    @InSequence(2)
+    @OperateOnDeployment("MicroProfileHealthApplicationStartupTestBase")
+    public void testApplicationStartupAfterDeployment(@ArquillianResource URL url) throws Exception {
+        try (CloseableHttpClient client = HttpClientBuilder.create().build()) {
+
+            checkGlobalOutcome(managementClient, "check-started", true, "myStartupProbe");
```

Comment:
Maybe the probe name can be refactored into a constant as you've done in the `manualmode` module parallel case?

Reply:
Thanks for addressing my comment. Although the check name is not a constant but still a literal, the main point about the refactoring was the class name, so I won't go further on this.
Feel free to resolve this thread.